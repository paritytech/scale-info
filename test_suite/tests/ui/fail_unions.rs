use scale_info::TypeInfo;

#[derive(TypeInfo)]
#[repr(C)]
union Commonwealth {
    a: u8,
    b: f32,
}

fn assert_type_info<T: TypeInfo + 'static>() {}

fn main() {
    assert_type_info::<Commonwealth>();
}
