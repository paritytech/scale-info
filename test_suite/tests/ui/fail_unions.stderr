error: Unions not supported
 --> $DIR/fail_unions.rs:4:1
  |
4 | / #[repr(C)]
5 | | union Commonwealth {
6 | |     a: u8,
7 | |     b: f32,
8 | | }
  | |_^

error[E0277]: the trait bound `Commonwealth: TypeInfo` is not satisfied
  --> $DIR/fail_unions.rs:13:24
   |
13 |     assert_type_info::<Commonwealth>();
   |                        ^^^^^^^^^^^^ the trait `TypeInfo` is not implemented for `Commonwealth`
   |
note: required by a bound in `assert_type_info`
  --> $DIR/fail_unions.rs:10:24
   |
10 | fn assert_type_info<T: TypeInfo + 'static>() {}
   |                        ^^^^^^^^ required by this bound in `assert_type_info`
